teps/0060-remote-resource-resolution.md

```diff
@@ -239,13 +242,106 @@ users can leverage those shared pipelines.
-nitty-gritty.
--->
+Prior to moving this TEP to `implementable` there will be a considerable amount
+of experimentation required to guage suitability of different approaches. There
+are a lot of open questions wrt syntax, approach and architecture that cannot
+be easily answered without trying ideas with running code.
+
+There are several additional changes that would aid experimentation in the
+pre-`implementable` phase of this TEP:
+
+### Proposal 1: Create an Experimental Resolution Project
+
+This proposal proposes adding a new controller in the [Experimental
+Repo](https://github.com/tektoncd/experimental/) that implements some of
+the ideas described in the [Alternatives section](#alternatives) of this TEP.
+
+This will need a new directory added with a controller & OWNERS configuration.
+If we want to try multiple alternatives (e.g. explore both interceptor-style
+and reconciler-style approaches) then we can either have separate subdirectories
+or branches for each implementation.
+
+### Proposal 2: Disable Tekton Pipelines' "In-Tree" Resolution Logic With a Deployment Arg
+
+A problem we face with introducing an experimental resolver controller is that
+any in-cluster process watching for `TaskRun` and `PipelineRun` resources will
+race with Tekton Pipelines to process them as they're submitted to the kube
+API.
+
+In order to support an experimental controller independently performing
+resolution we could add a behavioural flag that completely disables the
+built-in resolution logic. For example: `experimental-disable-ref-resolution`.
+
+This is explicitly _not_ being proposed as a feature flag because 1) it is not
+intended to migrate through to production-readiness and 2) a behavioural flag
+with its own name underlines the "experimental mode" that the user is electing
+to put their Tekton Pipelines controller in to.
+
+#### Alternative
+
+A different approach that would work for PipelineRuns today would be to require
+them to be created with a [`Pending`
+status](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#pending-pipelineruns).
+
+This proposal doesn't take that approach because:
+1. The same support does not exist for TaskRuns and so would require its own
+   independent TEP and review cycle to implement.
+2. Pending status can't be used by multiple controllers that all need to signal
+   readiness. This means a Resolution Reconciler and (for example) a custom
+   PipelineRun Scheduler could set this flag before the other controller is
+   ready to do so.
+
+### Proposal 3: Add `status.taskYaml` to `TaskRuns` and `status.pipelineYaml` to `PipelineRuns`
+
+When the proposed experimental resolution controller sees an unresolved
+`TaskRun` or `PipelineRun` it will read the resource and then resolve whatever
+ref, bundle or inline document is included. The resolved `spec` will then be
+embedded in the `status.taskSpec` field for `TaskRuns` or the
```

Comment:
Should this be `status.taskYAML.spec` and `status.pipelineYAML.spec`?

Reply:
Well spotted, thank you!  Fixed and pushed.